realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -1090,6 +1094,69 @@ private void notifyCallbackWithSuccess(RealmResults<PermissionOffer> permissions
         }
     }
 
+    private class RevokeOfferAsyncTask extends PermissionManagerTask<Permission> {
+
+        private final String offerToken;
+        private final RevokeOfferCallback callback;
+        private RealmResults<PermissionOffer> matchingOffers;
+
+        public RevokeOfferAsyncTask(PermissionManager permissionManager, String offerToken, RevokeOfferCallback callback) {
+            super(permissionManager, callback);
+            this.offerToken = offerToken;
+            this.callback = callback;
+        }
+
+        @Override
+        public void run() {
+            if (checkAndReportInvalidState()) {
+                return;
+            }
+            matchingOffers  = managementRealm.where(PermissionOffer.class)
+                    .equalTo("token", offerToken)
+                    .findAllAsync();
+            matchingOffers.addChangeListener(new RealmChangeListener<RealmResults<PermissionOffer>>() {
+                @Override
+                public void onChange(final RealmResults<PermissionOffer> offers) {
+                    if (offers.size() == 1) {
```

Comment:
Isn't `!offer.isEmpty()` enough?
I think we can remove all offers if there are multiple `PermissionOffer`s.